src/shapes/text.class.js

```diff
@@ -1132,38 +1145,65 @@
     },
 
     /**
-     * Turns the character into a 'superior figure' (aka. 'superscript')
-     * @param {Number} line the line number or starting position
-     * @param {Number} char the character number or final position
-     * @returns {Object} this
+     * Turns the character into a 'superior figure' (i.e. 'superscript')
+     * @param {Number} line the line number
+     * @param {Number} char the character number
+     * @returns {fabric.Text} thisArg
+     * @chainable
      */
     setSuperscript: function(line, char) {
-      return this._setScript(line, char, this.superscript);
+      return this._setScript(line, char, 'sup');
     },
 
     /**
-     * Turns the character into an 'inferior figure' (aka. 'subscript')
-     * @param {Number} line the line number or starting position
-     * @param {Number} char the character number or final position
-     * @returns {Object} this
+     * Turns the character into an 'inferior figure' (i.e. 'subscript')
+     * @param {Number} line the line number
+     * @param {Number} char the character number
+     * @returns {fabric.Text} thisArg
+     * @chainable
      */
     setSubscript: function(line, char) {
-      return this._setScript(line, char, this.subscript);
+      return this._setScript(line, char, 'sub');
     },
 
     /**
-     * Turns the character into an 'inferior figure' (aka. 'subscript')
-     * @param {Number} line the line number or starting position
-     * @param {Number} char the character number or final position
-     * @param {Object} schema either this.superscript or this.subscript
-     * @returns {Object} this
+     * Applies superscript/subscript at given position, according to active schema
      * @private
+     * @param {Number} line the line number
+     * @param {Number} char the character number
+     * @param {String} key one of {'sup', 'sub'}
+     * @returns {fabric.Text} thisArg
+     * @chainable
      */
-    _setScript: function(line, char, schema) {
+    _setScript: function(line, char, key) {
+      if (typeof key !== 'string') {
+        return this;
+      }
+
+      var schema = this.schema;
+      if (schema == null) {
+        return this;
+      }
+
+      var value = schema[key];
+      if (value == null) {
+        return this;
+      }
+
+      var size = value.size;
+      if (typeof size !== 'number') {
+        return this;
+      }
+
+      var baseline = value.baseline;
+      if (typeof baseline !== 'number') {
+        return this;
+      }
```

Comment:
the user is a developer. if it does it wrong, it will do it good after understanding whit it does not work. A nice jsdoc over the property will inform him what to use as schema or a sub/sup/superscript/subscript.